static/js/sub_store.ts

```diff
@@ -0,0 +1,68 @@
+import * as blueslip from "./blueslip";
+
+export type StreamSubscription = {
+    audible_notifications: boolean | null;
+    color: string;
+    date_created: number;
+    description: string;
+    desktop_notifications: boolean | null;
+    email_address: string;
+    email_notifications: boolean | null;
+    first_message_id: number | null;
+    history_public_to_subscribers: boolean;
+    in_home_view: boolean;
+    invite_only: boolean;
+    is_announcement_only: boolean;
+    is_muted: boolean;
+    is_web_public: boolean;
+    message_retention_days: number | null;
+    name: string;
+    pin_to_top: boolean;
+    push_notifications: boolean | null;
+    rendered_description: string;
+    role: 20 | 50;
```

Comment:
This should be [100, 200, 300, 400, or 600](https://github.com/zulip/zulip/blob/c99c423c81e5d8737011910a47acbce6066d5312/zerver/models.py#L1508-L1512).

Reply:
These should be just `20 | 50` based on the [`Subscription` model](https://github.com/zulip/zulip/blob/c99c423c81e5d8737011910a47acbce6066d5312/zerver/models.py#L2934-L2940). I think you mistakenly got these values from `UserProfile` model instead.

Reply:
Oh, right.

Should we make a `const enum` for this (and `stream_post_policy`)?

Reply:
Yeah, `const enum` sounds good. I also had to upgrade `@babel/preset-typescript` to get support for constant enums.